packages/infra/lib/ihe-gateway-v2-stack.ts

```diff
@@ -38,6 +44,54 @@ export class IHEGatewayV2LambdasNestedStack extends NestedStack {
     patientDiscoveryLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
     documentQueryLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
     documentRetrievalLambda.grantInvoke(props.apiService.taskDefinition.taskRole);
+
+    this.addMetricFiltersAndAlarms(patientDiscoveryLambda, "PatientDiscoveryLambda", props, [
+      { filterPattern: "Aborted Error In Soap Response", threshold: 100 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 100 },
+    ]);
+
+    this.addMetricFiltersAndAlarms(documentQueryLambda, "DocumentQueryLambda", props, [
+      { filterPattern: "RegistryErrorList In Soap Response", threshold: 5 },
+      { filterPattern: "Failure Sending SAML Request", threshold: 5 },
```

Comment:
I'd reconsider the filter pattern being so reliant on the full message. It's very prone to break as we maintain the code and potentially update the error message - no direct link to this code.

Maybe we could have some sort of fixed prefix that we add to all log entries we want to have a metric for, and that could be shared across the code and the infra using Core or Shared?